### DISCUSSION TITLE
Expose the currently active MetalK8s version behind a symbolic link

### DISCUSSION THREAD

**Discussion by gdemonet** [scality/metalk8s#3162] [Ideas]
To simplify automation and documentation of projects running on MetalK8s, it could be useful to provide a symbolic link to the mountpoint of the currently "active" version (same version as the `metalk8s.scality.com/cluster-version` annotation on the `kube-system` Namespace).

Projects could reference scripts using the symlink `/srv/scality/metalk8s`, which would resolve to `/srv/scality/metalk8s-{{current_version}}`:

```shell
# Add a new archive
/srv/scality/metalk8s/solutions.sh import --archive /path/to/my-archive.iso

# Activate this new archive
/srv/scality/metalk8s/solutions.sh activate --name my-solution --version 1.0.0
```

Such symlink could also be referenced in our own documentation, if desired. We could even imagine building "shim" scripts which can infer the desired version of a script to use depending on context. Take a `metalk8s-upgrade` shim:
- it would read the `destination-version` argument, and try to find the corresponding version in the available mounted ISOs
- if no such argument is provided, it would use the highest mounted version and upgrade from there

Many things are possible, but a first step to expose the current version would already make things simpler for admins trying to manage Solutions.

**Comment by NicolasT** [comment]
Not sure I'm super fond of this idea. It would require for us to be *very* sure we *always* keep that symlink 'correct' at all times. Especially on the second part, the upgrade script which 'magically' picks some version to upgrade to, I believe is dangerous. Better be explicit. It means one has to type a destination version (big deal!), but at least removes a bunch of potential surprises.

**Reply by gdemonet** [reply]
Agreed, that was my first concern as well.
I think building "shim" scripts which would check this version at runtime makes more sense than a symlink, actually.
As for upgrade, we could enforce that the shim requires a destination version argument.
